### PR TITLE
Page additions

### LINKED ISSUE #488: New Page additions
Based on the initial PR of #487 We need to update our MAINTAINERS.MD with this information to make sure we provide it to contributors that need assistance.

When a new page is created there are things that should be included.

- If a new data type is created then we need a new file in `_data/`. JSON is preferred but YAML/CSV is also okay.
- Add the route to `tests/conftest` in the `ROUTES` constant.

https://github.com/BlackPythonDevs/blackpythondevs.github.io/blob/56e317bdd1228ee25ee0f801f928178e0590e29f/tests/conftest.py#L14-L21

### PR DESCRIPTION
**Pull request by cmm25**
This pull request addresses two key areas:

1. **Comprehensive update to the `MAINTAINERS.md` file**
2. **Resolution of Ruby version incompatibility**

## MAINTAINERS.md Update

The `MAINTAINERS.md` file has been extensively revised to enhance clarity and accessibility for new and experienced contributors. Key updates include:

- Restructured content with a clear table of contents
- Added a "Getting Started" section to guide newcomers
- Provided detailed, step-by-step instructions for common tasks (e.g., accessibility testing, translations)
- Expanded sections on styling and adding new pages
- Simplified language while maintaining the depth of information
- Emphasized open communication and seeking help when needed

These improvements aim to streamline the contribution process and make it more approachable for contributors at all levels.

## Ruby Version Compatibility Fix

I encountered an error when trying to install Ruby 3.3.4:

```
rbenv: version `3.3.4' is not installed (set by /workspaces/blackpythondevs.github.io/.ruby-version)
```

To resolve this, the project has been updated to use Ruby 3.3.0, the latest available version in the `rbenv` list. The update includes:

1. Modifying the `.ruby-version` file to specify version 3.3.0
2. Adjusting the `Gemfile` for compatibility with Ruby 3.3.0
3. Updating the `Gemfile.lock` to reflect necessary dependency changes

These updates should resolve installation issues and provide a smoother setup experience for future contributors.

### CODE REVIEW

File: .ruby-version
Comment (kjaymiller) [inline .ruby-version:1]: I feel like we keep having problems with ruby version changing with development based on who is doing the development.

TODO: 

Either:
  Create Issue to add `.ruby-version` and `Gemfile.lock` to `gitignore`
or 
  Set ruby_version and gemfile.lock and do not allow updates to `.ruby_version` or `Gemfile.lock` unless that is the issue.

**Review by kjaymiller — APPROVED** [review]

**Review by kjaymiller — CHANGES_REQUESTED** [review]
If you can remove your ruby_version and gemfile.lock changes. 

After that it's good to go!

**Comment by kjaymiller** [comment]
@cmm25  see review. Only a minor issue and then we're all set to add.

**Comment by cmm25** [comment]
done check my lastest commit

**Review by kjaymiller — APPROVED** [review]

**Pull request merged**